### PR TITLE
214 add extra company information

### PR DESCRIPTION
**Pull request by CiscoPr**
8 tests are failing but the main code is done. Needs some reviews to discover where the errors are.

### CODE REVIEW

File: src/api/middleware/files.js [hunk 1]
```diff
@@ -99,3 +99,90 @@ export const cloudSave = async (req, res, next) => {
...
+            return next();
+        }
+    });
+};
+
+export const localSaveArray = async (req, res, next) => {
+    for (const file of req.files) {
+        const buffer = file.buffer;
+        const extension = file.mimetype.substr(file.mimetype.indexOf("/") + 1);
+        const filename = `${req.user.company}.${extension}`;
```
Comment (Naapperas) [inline src/api/middleware/files.js:138]: This line will make it so every file this company uploads has the same name. Maybe add a nonce or a sequence number to make the name unique.

**Review by Naapperas — CHANGES_REQUESTED** [review]

**Review by diogofonte — COMMENTED** [review]
The code and the tests implemented seem good to me, but the support for adding several locations is available? I didn't notice it in the files changes. I'm asking this because of this comment: https://github.com/NIAEFEUP/nijobs-be/issues/214#issuecomment-1307352719

**Comment by CiscoPr** [comment]
> The code and the tests implemented seem good to me, but the support for adding several locations is available? I didn't notice it in the files changes. I'm asking this because of this comment: [#214 (comment)](https://github.com/NIAEFEUP/nijobs-be/issues/214#issuecomment-1307352719)

yes i forgot to add that, thanks for reminding me.
@Naapperas, these several locations are also an optional company field or does it must have a minimal length?

**Comment by DoStini** [comment]
Just a note: commits should try to be small. On a more important note, commits should explain what it does or what it solves, and not the state of the issue. 

We also standardized using rebase instead of merge (from develop), if need help in this reach us

File: test/end-to-end/company.js [hunk 2]
```diff
@@ -441,6 +441,64 @@ describe("Company endpoint", () => {
...
+                        "location": "body",
+                        "msg": ValidationReasons.ARRAY_SIZE(CompanyConstants.locations.min_length, CompanyConstants.locations.max_length),
+                        "param": "locations",
+                        "value": locations
+                    });
+                });
+            });
+
+            describe("images", () => {
+                test("should return an error because the social media is too long", async () => {
```
Comment (pedronunes19) [inline test/end-to-end/company.js]: Shouldn't the text be about the images?

File: src/api/routes/company.js [hunk 3]
```diff
@@ -33,17 +33,24 @@ export default (app) => {
...
+        fileMiddleware.cloudSaveArray,
+        validators.finish,
         async (req, res, next) => {
 
             try {
                 const companyService = new CompanyService();
-                const { bio, contacts } = req.body;
+                const { bio, contacts, social, locations } = req.body;
                 const logo = req?.file?.url || `${config.webserver_host}/static/${req.file.filename}`;
+                const images = req?.files?.map((file) => file?.url || `${config.webserver_host}/static/${file.filename}`) ?? [];
```
Comment (Naapperas) [inline src/api/routes/company.js:46]: Perhaps this should be a function: should we need this logic elsewhere we only need to change it in one place.

**Review by Naapperas — CHANGES_REQUESTED** [review]

**Comment by Naapperas** [comment]
@CiscoPr What happened?

**Comment by CiscoPr** [comment]
@Naapperas i just created another  branch for this issue and I will open another PR with it. I wanted to start it over again to be less confusing since this one was really outdated

**Pull request closed without merging**